src/postgraphile/postgraphile.ts

```diff
@@ -114,41 +114,40 @@ export default function postgraphile(
   // must process them with `pluginHook` before we can rely on them.
   let incomingOptions: PostGraphileOptions;
 
-  // If the second argument is undefined, use defaults for both `schema` and
-  // `incomingOptions`.
-  if (typeof schemaOrOptions === 'undefined') {
-    schema = 'public';
-    incomingOptions = {};
-  }
   // If the second argument is a string or array, it is the schemas so set the
   // `schema` value and try to use the third argument (or a default) for
   // `incomingOptions`.
-  else if (typeof schemaOrOptions === 'string' || Array.isArray(schemaOrOptions)) {
+  if (typeof schemaOrOptions === 'string' || Array.isArray(schemaOrOptions)) {
     schema = schemaOrOptions;
     incomingOptions = maybeOptions || {};
   }
-  // Otherwise the second argument is the incomingOptions so set `schema` to the
+  // If the second argument is the incomingOptions so set `schema` to the
   // default and `incomingOptions` to the second argument.
+  else if (typeof schemaOrOptions !== 'undefined') {
+    schema = 'public';
+    incomingOptions = schemaOrOptions || {};
```

Comment:
```suggestion
    if (typeof schemaOrOptions !== 'object') {
      throw new Error("Expected schema string/array or options object, instead received " + typeof schemaOrOptions);
    }
    incomingOptions = schemaOrOptions || {};
```

This'll handle things like:

`postgraphile(pool, false)` / `postgraphile(pool, true)` / `postgraphile(pool, 0)` / `postgraphile(pool, true, {...})` / `postgraphile(pool, 0, {...})` I think.